traffic_ops/traffic_ops_golang/api/shared_handlers.go

```diff
@@ -315,25 +315,23 @@ func DeleteHandler(deleter Deleter) http.HandlerFunc {
 			HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
 			return
 		}
-		if !deleteKeyOptionExists {
```

Comment:
We can't remove this condition here without re-adding the logic elsewhere. Normal delete requires all all keys exist, so [`Identifier.GetKeyFieldsInfo()`](https://github.com/apache/trafficcontrol/blob/09bfc37828/traffic_ops/traffic_ops_golang/api/shared_interfaces.go#L50-L51) generally only contains a single key.

[`HasDeleteKeyOptions.DeleteKeyOptions()`](https://github.com/apache/trafficcontrol/blob/09bfc37828/traffic_ops/traffic_ops_golang/api/shared_interfaces.go#L106) is a separate array containing keys that *can* identify a resource, but [only requires](https://github.com/apache/trafficcontrol/blob/09bfc37828/traffic_ops/traffic_ops_golang/api/shared_handlers.go#L122-L129) that one such key exists.

`if !deleteKeyOptionExists {}` is used to keep the handler from exiting if `Identifier.GetKeyFieldsInfo()` keys do not all exist but at least 1 `HasDeleteKeyOptions.DeleteKeyOptions()` key does.

[`DELETE regions?name={{name}}`](https://traffic-control-cdn.readthedocs.io/en/latest/api/v3/regions.html#delete) currently fails with "missing key: id".

Reply:
Opened a PR to add an API test for the failing route: #4707